project/SimpleHTTPServer.scala

```diff
@@ -20,9 +20,19 @@ class SimpleHTTPServer(webDirectory: File, port: Int) extends Closeable {
   // needed for the future flatMap/onComplete in the end
   private implicit val executionContext = system.dispatcher
 
-  private val route = pathPrefix("") {
-    getFromDirectory(webDirectory.getAbsolutePath) ~ getFromFile(new File(webDirectory, "index.html"))
-  }
+  private val route =
+    getFromDirectory(webDirectory.getAbsolutePath) ~
+      // TODO: this needs improving. IT only supports one-folder deep fallback. It'd be good to have a generic fallback
+      // to serving 'index.html' from the appropriate folder anytime the path in the request leaves the last
+      // segment (filename) blank.
+      pathPrefix(Segment) { folderName =>
```

Comment:
Woot! Rereading the docs for more info on your suggestion I found `Segments` (in plural) does exactly what we need and it even supports empty paths. Please review my next commit for a double check but local tests worked as expected with `localhost:8080`, `localhost:8080/documentation`, `localhost:8080/blog` and `localhost:8080/blog` and even `localhost:8080/documentation/1.2.x/`. 

With `Segments` there's no need for a fallback either. 👍